src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -1763,6 +1763,42 @@ public SwitchParameter CNotMatch
             }
         }
 
+        /// <summary>
+        /// Get or sets binary operator -Imatchall.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = "MatchSet")]
+        [Alias("ImatchAll")]
+        public SwitchParameter ImatchAll
```

Comment:
This should be "IMatchAll". Other too.

I don't see "-Match" parameter for the Where-Object. 

And we need tests.

Reply:
I realize that all the binary operators for the Where-Object return True/False. However, the MatchAll operators return a collection of matches. It does not return a boolean type. It does not seem to me like the MatchAll operator is an appropriate operator for the Where-Object. Would you agree?

Reply:
I would agree. Generally, only comparison operators need be implemented in Where-Object (e.g., we don't have `Where-Object -Join` among others)

Reply:
I agree.